_rules/image-accessible-name-descriptive-qt1vmo.md

```diff
@@ -0,0 +1,295 @@
+		/>
+		<path fill="#e44d26" d="M107.6 471l-33-370.4h362.8l-33 370.2L255.7 512" />
+		<path fill="#f16529" d="M256 480.5V131H404.3L376 447" />
+		<path
+			fill="#ebebeb"
+			d="M142 176.3h114v45.4h-64.2l4.2 46.5h60v45.3H154.4M156.4 336.3H202l3.2 36.3 50.8 13.6v47.4l-93.2-26"
+		/>
+		<path fill="#fff" d="M369.6 176.3H255.8v45.4h109.6M361.3 268.2H255.8v45.4h56l-5.3 59-50.7 13.6v47.2l93-25.8" />
+	</svg>
+</html>
+```
+
+#### Failed Example 3
+
+This `canvas` element has an `aria-label` attribute that incorrectly describe the image.
+
+```html
+<html lang="en">
+	<canvas id="act" width="200" height="60" aria-label="HTML 5 logo"></canvas>
+	<script>
+		const ctx = document.querySelector('#act').getContext('2d')
+		ctx.font = '30px Arial'
+		ctx.fillText('ACT Rules!', 20, 40)
+	</script>
+</html>
+```
+
+### Inapplicable
+
+#### Inapplicable Example 1
+
+This `img` element has an empty (`""`) [accessible name][]. The image is described by an adjacent text.
+
+```html
+<img src="/test-assets/shared/pdf-icon.png" alt="" /> PDF document
+```
+
+#### Inapplicable Example 2
+
+This `img` element has an empty (`""`) [accessible name][] because of its `presentation` role.
+
+```html
+<html lang="en">
+	<img src="/test-assets/shared/fireworks.jpg" role="presentation" />
+</html>
+```
+
+#### Inapplicable Example 3
+
+This `svg` element has an empty (`""`) [accessible name][] because it has no attributes or content to provide an accessible name.
+
+```html
+<html lang="en">
+	<p>Happy new year!</p>
+	<svg height="200" xmlns="http://www.w3.org/2000/svg">
+		<polygon points="100,10 40,180 190,60 10,60 160,180" fill="yellow" />
+	</svg>
+</html>
+```
+
+#### Inapplicable Example 4
+
+This `canvas` element has an empty (`""`) [accessible name][] because it has no attributes or content to provide an accessible name.
+
+```html
+<html lang="en">
+	<p>Happy new year!</p>
+	<canvas id="newyear" width="200" height="200"></canvas>
+	<script>
+		const ctx = document.querySelector('#newyear').getContext('2d')
+		ctx.fillStyle = 'yellow'
+		ctx.beginPath()
+		ctx.moveTo(100, 10)
+		ctx.lineTo(40, 180)
+		ctx.lineTo(190, 60)
+		ctx.lineTo(10, 60)
+		ctx.lineTo(160, 180)
+		ctx.fill()
+	</script>
+</html>
+```
+
+#### Inapplicable Example 5
+
+This `img` element is not [visible][].
+
+```html
+<html lang="en">
+	<img src="/test-assets/shared/w3c-logo.png" alt="W3C" style="display:none" />
+</html>
+```
+
+#### Inapplicable Example 6
+
+This `canvas` element is not [visible][] because it is completely transparent.
+
+```html
+<html lang="en">
+	<canvas width="200" height="200"></canvas>
+</html>
+```
+
+#### Inapplicable Example 7
+
+This `img` element is has no [accessible name][] because it is not [included in the accessibility tree][].
+
+```html
+<html lang="en">
+	<img aria-hidden="true" src="/test-assets/shared/fireworks.jpg" alt="some picgure" />
```

Comment:
```suggestion
	<img aria-hidden="true" src="/test-assets/shared/fireworks.jpg" alt="some picture" />
```

Not that it makes any difference

Reply:
Actually, I'd still go for a descriptive name in that case. "some picture" is one of the less descriptive alt text possible…
```suggestion
	<img aria-hidden="true" src="/test-assets/shared/fireworks.jpg" alt="fireworks" />
```